packages/integration-functions/src/functions/appeal-document.js

```diff
@@ -75,6 +82,13 @@ function checkMessageIsValid(documentMessage, context) {
 	return isValid;
 }
 
+/**
+ * @param {import('@azure/functions').InvocationContext} context
+ * @returns {boolean}
+ */
+const documentShouldBeDeleted = (context) =>
+	context?.bindingData?.applicationProperties?.type === 'Delete';
```

Comment:
I've not got complete faith in this to be honest. It was inspired by https://github.com/Planning-Inspectorate/appeals-back-office/blob/main/appeals/functions/user-import/employee/index.js#L12-L27 but that function uses `import('@azure/functions').Context` as it's type, and `Context` doesn't seem to be in `@azure/functions`, at least not the version we're using

Reply:
https://learn.microsoft.com/en-us/azure/azure-functions/functions-node-upgrade-v4?tabs=v4&pivots=programming-language-javascript#review-your-usage-of-context

If this is just a property on the service bus message then think it's not on context but message

Reply:
I don't think it's on the message either, here's a sample on logged from the document function 
```
{
  documentId: 'ddd6b253-58fb-45bd-a0b0-830fd276a6b6',
  caseId: 556,
  caseReference: '6000556',
  version: 1,
  filename: 'Screenshot 2024-07-01 at 15.01.13.png',
  originalFilename: 'Screenshot 2024-07-01 at 15.01.13.png',
  size: 297121,
  mime: 'image/png',
  documentURI:
    'https://pinsstdocsappealsbotest.blob.core.windows.net/appeal/6000556/ddd6b253-58fb-45bd-a0b0-830fd276a6b6/v1/Screenshot 2024-07-01 at 15.01.13.png',
  publishedDocumentURI:
    'https://pinsstdocsappealsbotest.blob.core.windows.net/appeal/6000556/ddd6b253-58fb-45bd-a0b0-830fd276a6b6/v1/Screenshot 2024-07-01 at 15.01.13.png',
  virusCheckStatus: 'scanned',
  fileMD5: null,
  dateCreated: '2024-07-05T13:50:22.701Z',
  dateReceived: '2024-07-05T00:00:00.000Z',
  datePublished: '2024-07-05T13:50:22.701Z',
  lastModified: '2024-07-05T13:50:22.716Z',
  caseType: 'D',
  redactedStatus: 'not_redacted',
  documentType: 'caseDecisionLetter',
  sourceSystem: 'back-office-appeals',
  origin: 'pins',
  owner: null,
  author: null,
  description: null,
  caseStage: 'appeal-decision',
  horizonFolderId: null,
}
```

Reply:
Looks like this has changed as they've changed context in v4.

v3 has a `bindingData` property: [Context.d.ts#bindingData](https://github.com/Azure/azure-functions-nodejs-library/blob/v3.x/types/Context.d.ts#L28)

It looks like the equivalent in v4 is [InvocationContext.d.ts#triggerMetadata](https://github.com/Azure/azure-functions-nodejs-library/blob/v4.x/types/InvocationContext.d.ts#L90-L94)

I can't yet find any actual docs to tell us where these "custom properties" (as they are called in Service Bus Explorer) end up... we may have to log out the whole context or something to determine this. Perhaps with a deploy to dev from this branch.

Reply:
yep triggerMetadata was the one, thanks Ben!